rgf/utils.py

```diff
@@ -294,15 +300,28 @@ def _save_train_data(self, X, y, sample_weight):
             self._save_dense_files(X, y, sample_weight)
             self._is_sparse_train_X = False
 
-    def _execute_command(self, cmd, verbose=False):
-        output = subprocess.Popen(cmd,
-                                  stdout=subprocess.PIPE,
-                                  stderr=subprocess.STDOUT,
-                                  universal_newlines=True).communicate()
+    def fit(self, X, y, sample_weight):
+        self._set_paths()
+        if self.is_classification:
+            # Convert 1 to 1, 0 to -1
+            y = 2 * y - 1
```

Comment:
I compromised here.

Reply:
Don't you want to use there checks instead of your own ones?

https://github.com/scikit-learn/scikit-learn/blob/c75bccf43ecb2b6619491911f048676cab02aced/sklearn/base.py#L561

https://github.com/scikit-learn/scikit-learn/blob/c75bccf43ecb2b6619491911f048676cab02aced/sklearn/base.py#L577